data_steward/cdr_cleaner/clean_cdr_engine.py

```diff
@@ -140,6 +150,222 @@ def clean_dataset(project=None, statements=None, data_stage=stage.UNSPECIFIED):
+    """
+    destination_dataset_id = query_dict.get(cdr_consts.DESTINATION_DATASET)
+    destination_dataset_ref = gbq.DatasetReference(client.project,
+                                                   destination_dataset_id)
+    destination_table_id = query_dict.get(cdr_consts.DESTINATION_TABLE)
+    destination_table_ref = gbq.table.TableReference(destination_dataset_ref,
+                                                     destination_table_id)
+    return destination_table_ref
+
+
+def generate_job_config(client, query_dict):
+    """
+    Generates BigQuery job_configuration object
+
+    :param client: BigQuery client
+    :param query_dict: dictionary for the query
+    :return: BQ job_configuration object
+    """
+    job_config = gbq.job.QueryJobConfig()
+    if query_dict.get(cdr_consts.DESTINATION_TABLE) is None:
+        return job_config
+
+    destination_table = get_destination_ref(client, query_dict)
+
+    job_config.destination = destination_table
+    job_config.use_legacy_sql = query_dict.get(cdr_consts.LEGACY_SQL, False)
+    # allow_large_results can only be used if use_legacy_sql=True
+    job_config.allow_large_results = job_config.use_legacy_sql
+    job_config.write_disposition = query_dict.get(cdr_consts.DISPOSITION,
+                                                  bq_consts.WRITE_EMPTY)
+    return job_config
+
+
+def run_queries(client, query_list, module_info_dict):
+    """
+    Runs queries from the list of query_dicts
+
+    :param client: BigQuery client
+    :param query_list: list of query_dicts generated by a cleaning rule
+    :param module_info_dict: contains information about the query function
+    :return: integers indicating the number of queries that succeeded and failed
+    """
+    query_count = len(query_list)
+    successes, failures = [], []
+    for index, query_dict in enumerate(query_list):
+        try:
+            LOGGER.info(
+                ce_consts.QUERY_RUN_MESSAGE_TEMPLATE.render(
+                    index, query_count, **module_info_dict, **query_dict))
+            job_config = generate_job_config(client, query_dict)
+
+            module_name = module_info_dict[cdr_consts.MODULE_NAME]
+            query_job = client.query(query=query_dict.get(cdr_consts.QUERY),
+                                     job_config=job_config,
+                                     job_id_prefix=f'clean_{module_name}_')
+
+            # wait for job to complete
+            query_job.result()
+        except (GoogleCloudError, TOError) as exp:
+            LOGGER.exception(
+                ce_consts.FAILURE_MESSAGE_TEMPLATE.render(client.project,
+                                                          **module_info_dict,
+                                                          **query_dict,
+                                                          exception=exp))
+            failures.append(query_dict[cdr_consts.QUERY])
+            continue
```

Comment:
This change was due to migration from `bq_utils` to `utils/bq`. I referred to https://googleapis.dev/python/bigquery/latest/generated/google.cloud.bigquery.job.QueryJob.html#google.cloud.bigquery.job.QueryJob.result. Let me know if the earlier errors will suffice to catch these. I can also re-raise the error if you think it might be prudent.

Reply:
As discussed please re-raise